core/tests/protractor/embedding.js

```diff
@@ -161,33 +224,46 @@ describe('Embedding', function() {
       driver.get(
         general.SERVER_URL_PREFIX + general.SCRIPTS_URL_SLICE +
         'embedding_tests_dev_i18n_0.0.1.html');
-      browser.switchTo().frame(driver.findElement(by.xpath(
-          "//div[@class='protractor-test-embedded-exploration']/iframe")));
+
+      driver.findElement(
+        by.css('#embedding-exploration-id')).sendKeys(explorationId);
+      driver.findElement(
+        by.css('#embedding-submit-btn')).click();
+
+      browser.switchTo().frame(driver.findElement(
+        by.css('.protractor-test-embedded-exploration iframe')));
       general.waitForSystem();
       browser.waitForAngular();
       expect(driver.findElement(by.css('.protractor-test-float-form-input'))
         .getAttribute('placeholder')).toBe(expectedPlaceholder);
       browser.switchTo().defaultContent();
     };
 
-    users.createUser('embedder2@example.com', 'Embedder2');
+    users.createModerator('embedder2@example.com', 'Embedder2');
```

Comment:
Why moderator, and not a regular user?

Reply:
as this exploration was created by other user 'user1@embedding.com' in previous test so we need to give edit permissions.

Reply:
Oh gosh, no, I didn't realize that. Different tests need to be fully independent. If you change the order of the tests they should still run fine.

You might want to create a very small exploration for this new test instead, with a different username.

Reply:
@seanlip i have created a new test for second protractor test.